Adding docs on Tableau Cloud SAML integration (#1999)
FusionAuth/fusionauth-site: site/docs/v1/tech/samlv2/tableau-cloud.adoc

---
layout: doc
title: SAML v2 & Tableau Cloud
description: Setting up Tableau Cloud to login using FusionAuth via SAML v2
navcategory: login-methods
---

== Overview

> link:https://www.tableau.com/[Tableau] is one of the most advanced data visualization tools used for business intelligence and data analysis. This document covers the configuration necessary to get Tableau Cloud working with FusionAuth as the identity provider using SAML v2, allowing users to log in to your Tableau Cloud account using their FusionAuth credentials.

Author: Noted. Thanks.

=== Prerequisites

This document assumes you have a running instance of FusionAuth and a working Tableau Cloud application. You will also need admin accounts for both to configure them correctly.

> Finally, you'll need a FusionAuth user that you will use to sign into Zendesk. You can use an existing user or create a new user for this purpose.

Author: Sorry, bad copy and paste

=== Instructions

> We also recommend taking a look at link:https://help.tableau.com/current/online/en-us/saml_config_site.htm[Tableau Cloud SSO docs], but here are the instructions.

Author: > Can we submit this to them once it is published?

I've [posted a question in their forums](https://community.tableau.com/s/idea/0878b000000A2JHAA0/detail) and they said we could have it as a [KB article](https://www.tableau.com/support/knowledgebase) instead of the official docs.

> Why do we recommend them looking at the instructions?

I have this habit of having something like "please take a look at their docs" just in case they ever change their UI and our guide becomes outdated, but maybe I should put it in a Reference section or something. What do you think?

In your Tableau Cloud dashboard, navigate to [breadcrumb]#Settings -> Authentication#. Select [uielement]#Enable an additional authentication method#, choose [uielement]#SAML# and click [uielement]#Edit connection# to open the settings.

> image::samlv2/tableau/navigation.png[Steps to enable SAML in Tableau CLoud.,width=700,role=box-shadow bottom-cropped top-cropped]

Author: Don't you think it seems even more weird without the shadow?

![image](https://user-images.githubusercontent.com/1877191/224432621-de5be478-4bfa-4200-a93b-daebc609cfc6.png)

> image::samlv2/tableau/navigation.png[Steps to enable SAML in Tableau CLoud.,width=700,role=box-shadow bottom-cropped top-cropped]

Reviewer: I just took all the images again. With the screenshot scripts, we have a shadow added in the image, don't need the CSS class.

> Copy both [field]#Tableau Cloud entity ID# and [field]#Assertion Consumer Service URL (ACS)# from [uielement]#1. Export metadata from Tableau Cloud# and write them down. You'll use them shortly.

Reviewer: ```suggestion
Copy both [field]#Tableau Cloud entity ID# and [field]#Assertion Consumer Service URL (ACS)# from [uielement]#Export metadata from Tableau Cloud# and write them down. You'll use them shortly.
```

Reviewer: Let's get rid of all the numbers from the tableau navigation. I understand why you had them, but it looks weird.

> Now, log in to your FusionAuth instance to create a new application. Navigate to [breadcrumb]#Applications# and click the icon:plus[role=ui-button green,type=fas] button and give it a meaningful name (e.g. `Tableau`).

Reviewer: Are these icons available for all the FusionAuth admin ui icons? I like them (though they do cause some line height issues). If they are available, we should consider rolling them out across all the docs.

image::samlv2/tableau/config.png[Uploading metadata file.,width=494,role=box-shadow bottom-cropped top-cropped]

Click [uielement]#Test Connection# to open a modal with the FusionAuth login screen. Fill in your credentials and submit the form. You should see a [uielement]#Successfully connected to server# message below that button.

In [uielement]#5. Match attributes#, map email, first and last name fields like shown in the image below and click [uielement]#Apply#.

image::samlv2/tableau/mapping.png[Map email, first and last name fields.,width=476,role=box-shadow bottom-cropped top-cropped]

=== Log in

In your Tableau Cloud account, go to [breadcrumb]#Settings -> Authentication# (the same page you've configured the authentication above) and scroll down to [uielement]#Manage Users#. Click [uielement]#Manage Users# and then [uielement]#Add Users by Email# to send invites to the desired users. You must provide the same addresses they use to log in to your FusionAuth instance.

> After doing so, they should receive an email with the correct address to log in. When browsing to that page, they should enter their email address and then be redirected to the FusionAuth login screen after clicking [uielement]#Sign in#. There, they should fill in the credentials for their FusionAuth account and when logging in, they would finally arrive at the Tableau Cloud dashboard.

Reviewer: ```suggestion
After doing so, they should receive an email with a link to the login page. When browsing to that page, they should enter their email address. 

(can you put an image of the tableau screen here?)

They'll be redirected to the FusionAuth login screen when they click [uielement]#Sign in#. There, they should fill in the credentials for their FusionAuth account. After they authenticate successfully, they will be sent to the Tableau Cloud dashboard.

(image of tableau dashboard here if possible)
```

=== Troubleshooting

Make sure you pasted the right information from [field]#Tableau Cloud entity ID# and [field]#Assertion Consumer Service URL (ACS)# fields into your FusionAuth application.

> You can also browse to [breadcrumb]#Settings -> Authentication# in your Tableau Cloud dashboard and click [uielement]#Download log file# in [uielement]#7. Troubleshooting single sign-on (SSO)# to further investigate.

Reviewer: I don't think we need the numbers.
```suggestion
You can also browse to [breadcrumb]#Settings -> Authentication# in your Tableau Cloud dashboard and click [uielement]#Download log file# in [uielement]#Troubleshooting single sign-on (SSO)# to further investigate.
```
